### PR TITLE
Documentation cleanup

### PR DESCRIPTION
**Pull request by dnicolodi**
Mostly removes pages that have not yet been written from the index, and reorganizes the index a bit.

### CODE REVIEW

**Comment by rgommers** [comment]
This looks useful, thanks @dnicolodi. Not sure if you're still adding things? I'll review/merge if ready.

**Comment by dnicolodi** [comment]
I think the documentation would need some more love, but this is IMHO already a nice step forward while being minimally invasive. I'll open another PR for further work. I'll be happy if you could review this PR in its current form.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -10,13 +10,6 @@ on:
     branches:
       - main
       - release-*
-    paths-ignore:
```
Comment (rgommers) [inline .github/workflows/tests.yml:13]: I'd consider keeping this; even if it goes out of date I'd say it's better than triggering quite a bit of CI.

Comment (dnicolodi) [reply .github/workflows/tests.yml:13]: I updated the paths filter. Let's see if it works as intended...

**Comment by dnicolodi** [comment]
I've pushed two more commits with the suggested changes to the TODOs-only pages. PTAL.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -10,13 +10,14 @@ on:
...
+    paths:
       # This is spelt like this to do not ignore the docs/examples/ folder.
-      - docs/*
-      - docs/usage/*
-      - CHANGELOG.rst
-      - LICENSE
-      - README.md
+      - '**'
+      - '!docs/examples/**'
```
Comment (rgommers) [inline .github/workflows/tests.yml:16]: This didn't quite go right I think. It now ignores the examples and triggers on other files in `docs/`.

**Comment by rgommers** [comment]
Last two commits LGTM.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (dnicolodi) [reply .github/workflows/tests.yml]: Indeed. Fixed, I think.

**Review by rgommers — APPROVED** [review]

**Pull request merged**